crapssim/bet.py

```diff
@@ -25,22 +31,20 @@ class Bet(object):
 
     """
 
-    name = None
-    subname = ""
-    winning_numbers = []
-    losing_numbers = []
-    payoutratio = float(1)
-    # TODO: add whether bet can be removed
-
-    def __init__(self, bet_amount):
-        self.bet_amount = float(bet_amount)
-
-    # def __eq__(self, other):
-    #     return self.name == other.name
-
-    def _update_bet(self, table_object, dice_object: Dice):
-        status = None
-        win_amount = 0
+    def __init__(self, bet_amount: typing.SupportsFloat):
+        self.bet_amount: float = float(bet_amount)
+        self.name: str = str()
+        self.subname: str = str()
+        self.winning_numbers: list[int] = []
+        self.losing_numbers: list[int] = []
+        self.payoutratio: float = float(1)
+        self.removable: bool = True
+        self.can_be_placed_point_on = True
+        self.can_be_placed_point_off = True

```

Comment:
can we better represent these as 
```
self.bet_allowed: dict = {'On' : True, 'Off': True};
```

That way you can simply check,

```
if not bet.bet_allowed[table.point.status]:
    return
```

Reply:
Love this idea, the two variables that I had did feel clunky.

Initially I had toyed around with having a method for Bet that would do something similar where on the Base Class it would be
```
def bet_allowed(self, table_object: "Table") -> bool:
    return True
```

and then for point dependent bets override the method so for example PassLine would be
```
def bet_allowed(self, table_object: "Table") -> bool:
    if table_object.point.status == "on":
        return False
    return True
```

with my thinking being that if we ever wanted to do something where we add some more qualifiers to the Table like minimums/maximums or Table odds etc. we would be able to add to this method to determine whether the bet can be placed.

What I couldn't settle on was whether if the bets strategy amount was over the maximum should the player just bet up to the maximum amount, or should the player not make the bet at all so I never ended up doing anything with it and did the variables instead. 

Just thinking of this now too, if we ever added some more side bets we may want a method since something like a Fire bet requires a new shooter.

Reply:
I like that approach too but IMHO its better to  start simple and improve over time if & when needed.

Reply:
Added https://github.com/skent259/crapssim/pull/4/commits/ca472a9c2662eec63170859649070c336e22b52b

Reply:
See pull request #12 which added a method for this as well as a bet that requires that method (Fire.)

I decided to name the method just "allowed" as it felt redundant to me to type "bet.bet_allowed()".